core/background/pipeline/metadata.js

```diff
@@ -7,16 +7,30 @@ define([
 	return {
 		loadSong: function (song, cb) {
 
+			var isSongValid = false;
+
 			var onLoaded = function (isLfmValid) {
-				cb(isLfmValid);
+				isSongValid = isLfmValid;
 			};
 
-			let scrobbler = ScrobbleService.getFirstBound();
 
-			console.log('metadata loadSong() using: ' + scrobbler.getLabel());
+			let scrobblers = ScrobbleService.getAllBound();
+
+			for (var index in scrobblers) {
+				let scrobbler = scrobblers[index];
+				console.log('metadata loadSong() attempt using: ' + scrobbler.getLabel());
+
+				// load song metadata and update validation flag
+				scrobbler.loadSongInfo(song, onLoaded);
+
+				if (isSongValid) {
+					break;
```

Comment:
I'm not used to working with JS and dealing with it's callback madness. I'll take a look at your example when I have some time.